src/k8s/cmd/util/hooks.go

```diff
@@ -0,0 +1,99 @@
+package cmdutil
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"syscall"
+
+	"github.com/spf13/cobra"
+)
+
+const initialProcesEnvironmentVariables = "/proc/1/environ"
+
+// paths to validate if root in the owner
+var pathsOwnershipCheck = []string{"/sys", "/proc", "/dev/kmsg"}
+
+// HookCheckLXD checks ownership of dirs required for k8s to run.
+// HookCheckLXD if verbose true prints out list of potenital issues.
+// If potential issue found pops up warning for user.
+func HookCheckLXD(verbose bool) func(*cobra.Command, []string) {
+	return func(cmd *cobra.Command, args []string) {
+		if lxd, err := inLXDContainer(); lxd {
+			errMsgs := VerifyPaths()
+			if len(errMsgs) > 0 {
+				if verbose {
+					cmd.PrintErrln("Warning: When validating required resources potential issues found:")
+					for _, errMsg := range errMsgs {
+						cmd.PrintErrln("\t", errMsg)
+					}
+				}
+				cmd.PrintErrln("The lxc profile for Canonical Kubernetes might be missing.")
+				cmd.PrintErrln("For running k8s inside LXD container refer to " +
+					"https://documentation.ubuntu.com/canonical-kubernetes/latest/snap/howto/install/lxd/")
+			}
+		} else if err != nil {
+			cmd.PrintErrf(err.Error())
```

Comment:
question do we want to print out  to user issues with accessing `/proc/1/environ`? I'm thinking we might silently skip this error